app/uk/gov/hmrc/gform/obligation/ObligationService.scala

```diff
@@ -79,56 +111,42 @@ class ObligationService(gformConnector: GformConnector) {
     val hmrcTaxPeriodIdentifiers = formTemplate.expandFormTemplateFull.allFCs.collect {
       case IsHmrcTaxPeriod(el) => el
     }
-    val hmrcTaxPeriodIdentifiersNonEmpty: NonEmptyList[HmrcTaxPeriod] =
-      NonEmptyList.fromList(hmrcTaxPeriodIdentifiers) match {
-        case Some(x) => x
-        case _ =>
-          NonEmptyList(HmrcTaxPeriod(IdType("NONE"), TextExpression(Constant("NONE")), RegimeType("NONE")), List())
-      }
-
-    def ifStatement(condition: Boolean) =
-      if (condition) {
-        val newObligations =
-          lookupObligationsMultiple(formTemplate, authService, retrievals, form, hmrcTaxPeriodIdentifiersNonEmpty)
-        newObligations.map(i => form.copy(obligations = RetrievedObligations(i)))
-      } else {
-        Future.successful(form)
-      }
 
     hmrcTaxPeriodIdentifiers match {
-      case a :: _ =>
-        for {
-          currentIdNumber <- authService.evaluateSubmissionReference(
-                              a.idNumber,
+      case x :: xs =>
+        updateFormObligationsIfRequired(hmrcTaxPeriodIdentifiers, authService, retrievals, formTemplate, form)
+      case _ => Future.successful(form)
+    }
+  }
+
+  def updateFormObligationsIfRequired(
+    hmrcTaxPeriodIdentifiers: List[HmrcTaxPeriod],
+    authService: AuthService,
+    retrievals: MaterialisedRetrievals,
+    formTemplate: FormTemplate,
+    form: Form)(implicit hc: HeaderCarrier, ec: ExecutionContext) = {
+    val hmrcTaxPeriodIdentifiersNonEmpty = NonEmptyList(hmrcTaxPeriodIdentifiers.head, hmrcTaxPeriodIdentifiers.tail)
```

Comment:
`hmrcTaxPeriodIdentifiers.head` call is unsafe, we need to embrace the fact that in can fail.